odo/convert.py

```diff
@@ -213,14 +219,32 @@ def _():
             yield convert(pd.DataFrame, [], **kwargs)
     else:
         df = convert(pd.DataFrame, first, **kwargs)
+        df1, n1 = mkindex(df, 0)
 
         def _():
-            yield df
+            n = n1
+            yield df1
             for i in rest:
-                yield convert(pd.DataFrame, i, **kwargs)
+                df = convert(pd.DataFrame, i, **kwargs)
+                df, n = mkindex(df, n)
+                yield df
     return chunks(pd.DataFrame)(_)
 
 
+def _ignore_index(df, start):
+    return df, start
+
+
+def _add_index(df, start):
+    stop = start + len(df)
+    if hasattr(pd, 'RangeIndex'):
```

Comment:
Sorry I'm a little confused here. I figured I could do this...

    def _add_index(df, start, _idx_type=getattr(pd, 'RangeIndex', np.arange)):
        stop = start + len(df)
        idx = pd.Index(_idx_type(start=start, stop=stop))
        df.index = idx
        return df, stop

But if we do `_idx_type=getattr(pd, 'RangeIndex', pd.Index))` I think we'd need to check whether RangeIndex successfully imported. The arguments to RangeIndex and Index appear to be quite different... I'm able to avoid an `if` statement with this.